models/imap.go

```diff
@@ -0,0 +1,177 @@
+package models
+
+import (
+	"errors"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/glennzw/eazye"
+	log "github.com/gophish/gophish/logger"
+)
+
+// IMAP contains the attributes needed to handle logging into an IMAP server to check
+// for reported emails
+type IMAP struct {
+	UserId            int64     `json:"-" gorm:"column:user_id"`
+	Enabled           bool      `json:"enabled"`
+	Host              string    `json:"host"`
+	Port              uint16    `json:"port,string,omitempty"`
+	Username          string    `json:"username"`
+	Password          string    `json:"password"`
+	TLS               bool      `json:"tls"`
+	Folder            string    `json:"folder"`
+	RestrictDomain    string    `json:"restrict_domain"`
+	DeleteCampaign    bool      `json:"delete_campaign"`
+	LastLogin         time.Time `json:"last_login,omitempty"`
+	LastLoginFriendly string    `json:"last_login_friendly,omitonempty"`
```

Comment:
I'd recommend we remove this, personally. Instead, we can let the frontend (specifically moment.js) do all the work. This saves us from introducing another dependency and makes the backend logic cleaner.

Reply:
I initially tried to handle this purely in the frontend, but couldn't get moment.js to play nicely. It was a while ago now and I can't remember the specific issue but will dig into it again.

Reply:
The `LastLoginFriendly` database field isn't actually used to store data, it's defined in the IMAP struct so I can return the JSON value below:

```
ss[0].LastLoginFriendly = humanize.Time(ss[0].LastLogin)
delta := time.Now().Sub(ss[0].LastLogin).Hours() // Default value if never logged in is "0001-01-01T00:00:00Z"
if delta > 87600 {
	ss[0].LastLoginFriendly = "Never" // Well, either Never or > 10 years ago.
}
```
(See the code [here](https://github.com/glennzw/gophish/blob/imap/controllers/api/imap.go#L46))

I'm not sure if there's a way to return additional fields without having them defined in the imap struct?

The reason I ended up doing it this way was: (i) to avoid time differences between client local time and server time (including thinking about timezones),  (ii) it was easier to handle on the server side, (iii) `moment.js` doesn't have granularity below 44 seconds:

<img width="738" alt="Screen Shot 2019-10-28 at 13 33 57" src="https://user-images.githubusercontent.com/3966613/67682599-a96e7380-f987-11e9-8a65-0b572139ce82.png">
 
So two questions maybe someone can help with:

- Can I include a JSON response key/pair that's not defined in the IMAP struct
- Do you know how to wrangle `moment.js` to have greater granularity below 44 seconds?
- Rather, if you think the `moment.js` granularity is fine we can just use that

Alternatively @jordan-wright if you don't think this is useful I can remove it. I found it useful to have in the UI to know if Gophish was successfully logging in when configuring my IMAP server.